### PR TITLE
Add event types: create, delete, repository, and remove travis-sidekiqs

### PR DESCRIPTION
**Pull request by svenfuchs**
Add a few more event types, use Sidekiq::Client.push, get rid of the extra dependency, allow setting the queue name via config or env.

Re https://github.com/travis-pro/team-teal/issues/868. This should wait until https://github.com/travis-ci/travis-gatekeeper/pull/82 is in production.

### CODE REVIEW

**Comment by svenfuchs** [comment]
@henrikhodne I've tested this on org staging, and everything seems to be working as expected:

https://github.com/travis-pro/team-teal/issues/1354#issuecomment-250139914

File: lib/travis/listener/app.rb [hunk 1]
```diff
@@ -80,15 +82,15 @@ def handle_event
...
+        Travis::Gatekeeper.push(Travis.config.gator.queue, data)
       end
 
       def handle_event?
         settings.events.include?(event_type)
       end
 
       def log_event(event_details, event_basics)
-        info(event_details.merge(event_basics).map{|k,v| "#{k}=#{v}"}.join(" "))
+        info(event_basics.merge(event_details).map{|k,v| "#{k}=#{v}"}.join(" "))
```
Comment (henrikhodne) [inline lib/travis/listener/app.rb]: Out of curiosity, why were these swapped?

Comment (henrikhodne) [inline lib/travis/listener/app.rb]: Nope, that makes lots of sense to me, I forgot that Hashes are ordered in Ruby... I was just a bit surprised and was curious what the difference was, thanks for explaining! :heart:

**Comment by svenfuchs** [comment]
Shipped

**Pull request merged**